### PR TITLE
Added support for LedgerEntry classes, SorobanResources and SorobanTransactionData

### PR DESCRIPTION
**Pull request by cuongph87**
- Added support for LedgerEntry classes, SorobanResources and SorobanTransactionData.
- Renamed class Signer to SignerUtil.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have read the **CONTRIBUTING** document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: stellar-dotnet-sdk/TrustlineAsset.cs [hunk 1]
```diff
@@ -122,9 +123,9 @@ public override xdr.TrustLineAsset ToXdr()
                 trustlineAssetXdr.Discriminant = assetXdr.Discriminant;
                 trustlineAssetXdr.AlphaNum4 = assetXdr.AlphaNum4;
                 trustlineAssetXdr.AlphaNum12 = assetXdr.AlphaNum12;
-
+                // TODO PoolID implementation
```
Comment (jopmiddelkamp) [inline stellar-dotnet-sdk/TrustlineAsset.cs:126]: Is it a lot of work to add this into the PR?

**Review by jopmiddelkamp — CHANGES_REQUESTED** [review]

File: stellar-dotnet-sdk/ConfigSettingContractBandwidthV0.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+﻿using stellar_dotnet_sdk.xdr;
+
+namespace stellar_dotnet_sdk;
+
+public class ConfigSettingContractBandwidthV0 : LedgerEntryConfigSetting
```
[the remark this answers is not in the document]
Comment (cuongph87) [reply stellar-dotnet-sdk/ConfigSettingContractBandwidthV0.cs:5]: I thought about it, but wasn't sure if they would add V1 at some point.

File: stellar-dotnet-sdk-test/LedgerEntryTest.cs [hunk 3]
```diff
@@ -0,0 +1,1133 @@
...
+        };
+    }
+
+    private LedgerEntryContractData InitBasicLedgerEntryContractData()
+    {
+        return new LedgerEntryContractData
+        {
+            LastModifiedLedgerSeq = 10000,
+            LedgerExtensionV1 = _ledgerExtensionV1,
+            Key = new SCString("key 1"),
```
[the remark this answers is not in the document]
Comment (cuongph87) [reply stellar-dotnet-sdk-test/LedgerEntryTest.cs:118]: No, we can't, as the "generated" Key and Value fields are SCVal, not SCString/SCUint64.

File: stellar-dotnet-sdk-test/LedgerEntryTest.cs [hunk 4]
```diff
@@ -0,0 +1,1133 @@
...
+    public void TestLedgerEntryAccountWithAccountExtensionV1AndV2()
+    {
+        var ledgerEntry = InitBasicLedgerEntryAccount();
+
+        var signerSponsoringIDs = new[] { _signerSponsoringId };
+
+        ledgerEntry.AccountExtensionV1 = new AccountEntryExtensionV1
+        {
+            Liabilities = new Liabilities(100, 200),
+            ExtensionV2 = new AccountEntryExtensionV2
```
[the remark this answers is not in the document]
Comment (cuongph87) [reply stellar-dotnet-sdk-test/LedgerEntryTest.cs:207]: I actually thought about it, was wondering if they wanted to "isolate" the extensions in the original schemes for a reason, beside memory optimization.

File: stellar-dotnet-sdk-test/TransactionTest.cs [hunk 5]
```diff
@@ -588,5 +566,61 @@ public void TestTransactionFeeOverflow()
...
+                    WriteBytes = 30
+                }
+            };
+            var tx = new TransactionBuilder(account)
+                .SetFee(100)
+                .AddTimeBounds(new TimeBounds(0, 0))
+                .AddOperation(
+                    new PaymentOperation.Builder(destination, asset, amount).Build())
+                .AddMemo(new MemoText("Happy birthday!"))
+                .SetSorobanTransactionData(sorobanData)
```
[the remark this answers is not in the document]
Comment (cuongph87) [reply stellar-dotnet-sdk-test/TransactionTest.cs:605]: The Dart SDK doesn't have support for those yet. But for almost every test, I always try to decode the XDR base64 string in the Lab to check it's valid or not. Do you think it's relevant?

**Review by jopmiddelkamp — APPROVED** [review]

**Pull request merged**